### ISSUE TITLE
Module update stays active despite update completed

### ISSUE THREAD

**Issue by selwynorren** [thirtybees/thirtybees#1747]
On a brand new install of Thirtybees 1.5.1 it shows that 3 modules need upgrading. I check module version number on server before running the update.

Update completed, as there is a version number change, however in the admin, it still shows that the module needs updating

**Comment by selwynorren** [comment]
I have now updated my other store to 1.5.1 for PHP 8.1 and now this store also has the same issue

**Comment by getdatakick** [comment]
I can't reproduce this on my environment. It's possible that you have some override (Module.php class most likely) that does not work properly in 1.5.x

Can you please provide list of your existing overrides?

**Comment by selwynorren** [comment]
Howdy @getdatakick So this is a default install of Thirtybees where I detected the issue originally. So there are no overrides at all. I have also just conformed that, and there are no overrides at all.

my other installation dies indeed have the nocaptcha recaptcha module installed and that is the only override in place.

As mentioned when I downgrade the server to PHP 7.4, then both work no issues. So I am running PHP 8.1 with Apache FPM on both. So when I take the server back to 8.1 then the error starts again.

I also made sure I downloaded the v1.5.1 specifically for PHP 8.1

Super strange that it does indeed run the updates, so this is not an issue, just in the back office, it does not update the fact that the module updates were successful. so the back office still reads the old version number, however the files on the server have all been updated.

This is most probably why there are no error logs as teh action was successful

**Comment by getdatakick** [comment]
Thanks for additional information.

This now looks like PHP 8.1 caching issues, probably opcache. Check your PHP opcache configuration, especially [opcache.revalidate_freq](https://www.php.net/manual/en/opcache.configuration.php#ini.opcache.revalidate-freq)

If set incorrectly, it can take your server some time to detect changes to php files on filesystem, and will still use old version from the cache memory. This can cause a lot of troubles during update -- instead of running update scripts on new version of module, it runs on old version of module.

**Comment by selwynorren** [comment]
Thanks @getdatakick So I don't have opcache installed on either of these two servers. I also thought initially it was a caching issue, but I dont have any server side caching on one of the servers (The clean installed Thirtybees).

I have also restarted teh whole server, as well as apache and fpm and I still have the same issue

**Comment by getdatakick** [comment]
Can you give me access to your admin so I could have a look?

**Comment by selwynorren** [comment]
absolutely, can I send to you in a DM? of would you prefer some other method?

**Comment by WBNet-Wout** [comment]
I am having the same issue with a clean install on Debian bookworm using PHP 8.2.

When I e.g. try to update the Core Updater module, it will say:

>De volgende module(s) werd(en) met success bijgewerkt:
>-    Core Updater :
>        Huidige versie: 1.6.5

While 1.6.5 is the old version I had, so it actually hasn't updated. In e.g. `/var/www/thirtybees/modules/coreupdater/config_nl.xml` I then also still see: `<version><![CDATA[1.6.5]]></version>`.

I do however not have this issue with my production install on Debian bullseye using PHP 7.4.

**Comment by getdatakick** [comment]
This is most likely related to php opcache. Some servers are set to not clear opcache when php file changed. So even though there is new php version no disk, php server is still using cached version of module class.

**Comment by WBNet-Wout** [comment]
I doubt that. After setting `opcache.enable=0` in `/etc/php/8.2/fpm/php.ini` and restarting PHP-FPM I still have this issue.

**Comment by selwynorren** [comment]
I have some more information about this. I posted a lengthy answer in the forums over here:
https://forum.thirtybees.com/topic/6426-module-updates-but-still-shows-need-to-update/

TL:DR
The config.xml file does not get updated with the latest version number, even though the modules and all files are updated. After a few tests, I noticed even the database does not get updated with the correct version number, however after deleting the config.xml file, the module updates with the correct version number in the back-office, the database also gets updated, and a new config.xml file get downloaded/created. However if you manually edit that file to reflect an older module version number, we are back to square one again.

Issue lies in not being able to over write the config.xml file. I looked for an error log reflecting the same, however nothing is produced

**Comment by getdatakick** [comment]
Should be fixed by 30e0072ee6a99096250d14631a7f365c86477cdf
